vowpalwabbit/core/src/reductions/cb/cb_explore.cc

```diff
@@ -333,31 +333,44 @@ void generic_output_example(VW::workspace& all, float loss, VW::example& ec, CB:
   print_update_cb_explore(all, CB::is_test_label(ld), ec, sso);
 }
 
-}  // namespace CB_EXPLORE
+struct options_cb_explore_v1
+{
+  uint32_t num_actions;
+  uint64_t tau;
+  float epsilon;
+  uint64_t bag_size;
+  uint64_t cover_size;
+  bool nounif;
+  float psi;
+  bool cover_supplied;
```

Comment:
Since there's only one valid configuration for these options, I suggest an enum class

```
enum class cb_explore_types
{
  GREEDY,
  COVER,
  COVER_WITH_EPSILON,
  BAG,
  FIRST
}
```

That will programmatically restrict the reduction to its only valid options (default will be GREEDY, then overwrite with something else if necessary)

Reply:
When serialized, enums cause an unusual problem.  In most serialize/deserialize protocols, it's not possible to deserialize a valid enum when new items are added, even it is to check for error.  I started out defining enums for serializable fields and then had to switch to integers for this reason.  

Defining enums and setting it to a serializable int is ok though.